kyaml/openapi/openapi.go

```diff
@@ -44,6 +45,49 @@ func SchemaForResourceType(t yaml.TypeMeta) *ResourceSchema {
 	return &ResourceSchema{Schema: rs}
 }
 
+// AddSchemaFromFile reads the file at path and parses the OpenAPI definitions
+// from the field "openAPI"
+func AddSchemaFromFile(path string) error {
+	b, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
+	}
+
+	// parse the yaml file (json is a subset of yaml, so will also parse)
+	y, err := yaml.Parse(string(b))
+	if err != nil {
+		return err
+	}
+	// get the field containing the openAPI
+	f := y.Field("openAPI")
+
+	defs, err := f.Value.String()
+	if err != nil {
+		return err
```

Comment:
Should we return error here? What if Kptfile is empty and there are no setters yet. It is erroring out.

Reply:
good catch, `f` would be nil if there were no OpenAPI definitions.  fixed